MCMC/simulation.py

```diff
@@ -47,7 +48,7 @@ def __init__(
         self.hard_sphere = hard_sphere
         self.kwargs = kwargs
         self.system_history = [np.copy(self.system)]
-        self.energies = [np.copy(self.energy)]
+        self.energies = [copy.deepcopy(self.energy)]
```

Comment:
What does deep copy do in comparison to just copy? Is this how you're saving it as an array instead of value?

Reply:
deep copy creates a replica of the variable self.energy. The reason for using a copy operation is that in Python assignment statements do not entirely copy the object. It just creates a new variable and uses the same reference as the original object. The downside is that any changes in the original object will reflect on the new variable. It means that if `self.energy` changes during simulation, the first item in `self.energies` list will change too. 
In order to prevent that, we use python deep copy to create a "real" copy of the original variable, so that we make sure it doesn't change. 
Check [this link ](https://www.geeksforgeeks.org/copy-python-deep-copy-shallow-copy/)for some cool examples on shallow and deep copy.